### PR TITLE
add test functions in the test_utils.py and some test data

### PR DESCRIPTION
**Pull request by cirrusasf**

### CODE REVIEW

**Review by forrestfwilliams — COMMENTED** [review]
Left some comments on some of the tests, but overall looking good.

I tried running the merge workflow for two burst insar pairs from this branch and unfortunately it failed. I checked, and it is working from the `merge_bursts` branch. Could you track down the issue? Could be something related to the `create_image` changes.

File: tests/test_utils.py [hunk 1]
```diff
@@ -171,3 +202,160 @@ def test_get_esa_credentials_missing(tmp_path, monkeypatch):
         msg = 'Please provide.*'
         with pytest.raises(ValueError, match=msg):
             get_esa_credentials()
+
+
+def test_create_image(tmp_path):
+    def _check_create_image(path: str, image_subtype: str = 'default'):
+        # test ifg in create, finalize, and load
+        path_c = path + '/img_c'
```
[the remark this answers is not in the document]
Comment (cirrusasf) [reply tests/test_utils.py:208]: yes, it indicates create, finalize, and load, respectively. I will make them more clear.

**Comment by forrestfwilliams** [comment]
Hey @cirrusasf, running `merge_tops_bursts` still fails from this branch. Can you revert the changes to `utils.create_image` to see if that fixes things?

**Comment by cirrusasf** [comment]
> Hey @cirrusasf, running `merge_tops_bursts` still fails from this branch. Can you revert the changes to `utils.create_image` to see if that fixes things?

problem solved.

File: tests/__init__.py
Comment (forrestfwilliams) [inline tests/__init__.py:1]: We don't actually need this if we just import `isceobj` after `hyp3_isce2`. I think we should delete this file. See my suggestion in `test_utils.py` for more.

File: tests/test_utils.py [hunk 2]
```diff
@@ -1,20 +1,34 @@
+import filecmp
 import os
+import shutil
+from pathlib import Path
 
+import isceobj
```
Comment (forrestfwilliams) [inline tests/test_utils.py:6]:
```suggestion
```

File: tests/test_utils.py [hunk 3]
```diff
@@ -1,20 +1,34 @@
...
+    load_isce2_image,
     make_browse_image,
     oldest_granule_first,
+    read_product_metadata,
     resample_to_radar,
+    resample_to_radar_io,
     utm_from_lon_lat,
+    write_isce2_image,
+    write_isce2_image_from_obj,
 )
```
Comment (forrestfwilliams) [inline tests/test_utils.py:30]:
```suggestion

import isceobj  # noqa: F401
```

File: tests/test_utils.py [hunk 4]
```diff
@@ -1,20 +1,34 @@
...
 import os
+import shutil
+from pathlib import Path
 
+import isceobj
 import numpy as np
 import pytest
 from osgeo import gdal
 
 from hyp3_isce2.utils import (
```
Comment (forrestfwilliams) [inline tests/test_utils.py:11]: we're basically importing the entire `utils` module here. Can we convert this to `from hyp3_isce2 import utils` and just modify the function calls?

Comment (cirrusasf) [reply tests/test_utils.py:11]: will do it.

File: tests/__init__.py
Comment (cirrusasf) [reply tests/__init__.py:1]: if import isceobj after hyp3_isce2, it can not pass the flake8 test.

File: tests/test_utils.py [hunk 5]
```diff
@@ -1,19 +1,14 @@
...
-    ESA_HOST,
-    GDALConfigManager,
-    extent_from_geotransform,
-    get_esa_credentials,
-    make_browse_image,
-    oldest_granule_first,
-    resample_to_radar,
-    utm_from_lon_lat,
-)
+from hyp3_isce2.utils import *
```
Comment (forrestfwilliams) [inline tests/test_utils.py:10]:
```suggestion
import hyp3_isce2.utils as utils
```
This is better practice and Flake8 will not pass without this change. Unfortunately, you will have to update the references to utils throughout the file.

**Review by forrestfwilliams — APPROVED** [review]
LGTM!

**Pull request merged**